src/dataset.jl

```diff
@@ -306,7 +423,7 @@ function create(
     width::Integer = 0,
     height::Integer = 0,
     nbands::Integer = 0,
-    dtype::DataType = Any,
+    dtype = GDT_Unknown,
```

Comment:
`create` calls `GDAL.gdalcreate` which expects a `GDALDataType` which is an enum, therefore an int, and not a datatype if I get it right. There seems to be `convert` defined for conversion of `Any` and number types to `GDALDataType. So the way it was is fine, but after this change you can either specify the julia number type or the gdal data type directly. I now added the change to `unsafe_create` as well.

Reply:
Ohh I see. Can we add a annotation for `dtype`, like `dtype::Union{DataType, GDALDataType}` in that case?

Personally I'd prefer to not expose `GDALDataType` (since it doesn't provide any benefit, it's an extra symbol for the package to maintain, and it confuses the user to have to make that choice), but I don't feel strongly about it.

Reply:
I think as this change is not related to the vector writer I might as well revert it. The other option would be to just remove the type annotation and set the default to `Any`, but you like to have the type annotations for clarity, right?